etc/deltaDebug.py

```diff
@@ -187,6 +193,11 @@ def debug(self):
                 if err is None or self.get_cuts() == 0:
                     break
 
+        # Destroy the DB in memory to avoid being out-of-memory when
+        # the step code is running
+        if (self.base_db is not None):
+            self.base_db.destroy(self.base_db)
```

Comment:
Rather than moving the destroy call out of perform_step I think it would be better to capture the relevant info inside that step (num insts & nets) and return it to the caller.  Moving it here still increases the high memory point.